### ISSUE TITLE
status is missing in SearchResult

### ISSUE THREAD

**Issue by PaulDennetiere** [olivere/elastic#1183]
Please use the following questions as a guideline to help me answer
your issue/question without further inquiry. Thank you.

### Which version of Elastic are you using?

[ ] elastic.v7 (for Elasticsearch 7.x)
[x] elastic.v6 (for Elasticsearch 6.x)
[ ] elastic.v5 (for Elasticsearch 5.x)
[ ] elastic.v3 (for Elasticsearch 2.x)
[ ] elastic.v2 (for Elasticsearch 1.x)

### Please describe the expected behavior
I'm using the msearch feature to perform between 1 and 3 searches at a time, depending on business rules.
While i can loop through the results and get the ErrorDetails from every result, Iz would like to have the status of those errors (eg http status like 400 for a bad request)

Is it a voluntary omission ?

Looking in other issue i understand that i could use PerformRequest instead and then parse the result myself, but it seems unnatural.

Am I missing something ?

**Comment by olivere** [comment]
Thanks for the report. I suppose you're referring to the HTTP status with [Multi Get API](https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-multi-get.html), right?

In case of an error, you will not get the response, but an error. The error is of type [`*elastic.Error`](https://github.com/olivere/elastic/blob/release-branch.v7/errors.go#L66-L70). To check if there was a specific HTTP status, you can use [`elastic.IsStatusCode(err interface{}, code int)`](https://github.com/olivere/elastic/blob/release-branch.v7/errors.go#L148-L163).

You're right in that there is no helper to get _just_ the HTTP status. Is that what you're looking for?

As a workaround, you can also write your own [`Decoder`](https://github.com/olivere/elastic/wiki/Decoder), which is invoked before parsing each response from ES.

**Comment by PaulDennetiere** [comment]
Hi, first thanks for your quick reply.

No i'm talking about the [Multi Search API](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-multi-search.html). I understand that :
```
msearch := elastic.MultiSearch()
search1 := elastic.NewSearchRequest().Query(whatever)
search2 := elastic.NewSearchRequest().Query(whatever)
msearch.Add(search1, search2)
results, err := msearch.Do(ctx)
```
in this case, I will have a global error (err) if something went wrong with the network (unreachable, timeout etc..). But I'd like to have a status for each result from `results.Responses`
For exemple I could have a 400 bad request from search1 and 200 from search2. In such a case err will be nil, and `results.Responses[0].Error` will be filled with a `elastic.ErrorDetails` 
I find it unnatural to parse the cause description contained in `elastic.ErrorDetails`, while elasticsearch provide a status for each result (as you can see by performing a multisearch from kibana or a curl) 
I tried locally to add the Field `Status string` to the SearchResult struct, and it does what I expected: add a status for each response.

That's why I'm asking if this is a known omission for some reason I don't get, or if I am missing something that have an equivalent behavior. 

I did look at the Decoder, however I'm confortable with using the standard json unmarshaller. But I can't change what struct is passed to the decoder at response time (which is a MultiSearchResult)

**Comment by olivere** [comment]
Sorry for the delay.

I’ve looked up the Java code and in fact there’s a Status being rendered for each response item (https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/action/search/MultiSearchResponse.java#L174). We will need to add it.

There is no reason why that field is missing other than lack of resources to keep up with the pace of changes in ES. I guess that field simply wasn’t there when implementing MultiSearch API.

**Comment by olivere** [comment]
I've [added `Status` here](https://github.com/olivere/elastic/blob/release-branch.v7/search.go#L535) for the upcoming 7.0.7 release, in 0f7dc056dd67cd25d912716b7dc288ca916726a7.

I need to backport it to 6.x though.

**Comment by olivere** [comment]
Should be available in 6.2.24 and 7.0.7, once it is published.
